massa-protocol-worker/src/protocol_worker.rs

```diff
@@ -148,9 +149,11 @@ pub struct ProtocolWorker {
     /// List of processed headers
     checked_headers: Map<BlockId, BlockInfo>,
     /// List of ids of operations that we asked to the nodes
-    pub(crate) asked_operations: HashMap<OperationId, (Instant, Vec<NodeId>)>,
+    pub(crate) asked_operations: HashMap<OperationPrefixId, (Instant, Vec<NodeId>)>,
     /// Buffer for operations that we want later
     pub(crate) op_batch_buffer: OperationBatchBuffer,
+    /// Adapter to retreive full operation ids from a given prefix
+    pub(crate) op_prefix_adapter: OperationIdAdapter,
```

Comment:
I think instead of adding a new structures, you can replace `checked_operations: OperationIds` with something like a `checked_operations: OperationIdAdapter`, which could be renamed to `CheckedOperations`.